s3_bucket.go

```diff
@@ -433,7 +436,37 @@ func (s *s3Bucket) Get(ctx context.Context, key string) (io.ReadCloser, error) {
 	return s.Reader(ctx, key)
 }
 
-func uploadHelper(ctx context.Context, b Bucket, key, path string) error {
+type fundamentalBucket interface {
+	Put(context.Context, string, io.Reader) error
+	Get(context.Context, string) (io.ReadCloser, error)
+
+	Writer(context.Context, string) (io.WriteCloser, error)
+	Reader(context.Context, string) (io.ReadCloser, error)
+	Remove(context.Context, string) error
+}
+
+func (s *s3Bucket) s3WithUploadChecksumHelper(ctx context.Context, bucketName string, svc *s3.S3, target, file string) (bool, error) {
```

Comment:
This doesn't need to be a method on s3Bucket if the `bucketName` and `svc` arguments shouldn't be passed in. Why are the explicitly passed in?

Reply:
good call!